lldb/source/Target/StackFrameList.cpp

```diff
@@ -936,3 +936,7 @@ size_t StackFrameList::GetStatus(Stream &strm, uint32_t first_frame,
   strm.IndentLess();
   return num_frames_displayed;
 }
+
+void StackFrameList::ClearSelectedFrameIndex() {
```

Comment:
In StackFrameList::Clear we take the m_list_mutex before clearing the selected frame (among other things).  Should we also do that here?

Reply:
hmmmm good question. `GetSelectedFrameIndex` is not taking the mutex either. On first glance I would've thought the mutex is there to just protect the `m_frames`. There's a separate `m_inlined_depth_mutex` that protects the corresponding member (for performance reason based on the comment next to it).

But if `ClearSelectedFrameIndex` and `GetSelectedFrameIndex` they should be protected against concurrent access. Do we have concerns with using the list mutex for that?

Reply:
`GetSelectedFrameIndex` modifies `m_selected_frame_idx`. So concurrent access would already be an issue there. Maybe we're just getting away with it now (because it just doesn't happen much in practice)?

Reply:
This is a shared mutex so it can't be recursive, so we'd need to watch out for that.

Reply:
Hmm we also write to `m_selected_frame_idx` in `StackFrameList::SetSelectedFrame`, which holds a `shared_lock`. I think after the refactor to make the lock non-recursive, we lost some of protection against concurrent modifications to `m_selected_frame_idx`. Probably because it wasn't ever clear what `m_list_mutex` is supposed to protect (the entire class or just the frames collection).

And we end up calling `GetSelectedFrameIndex` from `SetSelectedFrame`, so we would need to recursively lock. Unless introduce a separate mutex for `m_selected_frame_idx`, which I'm not a big fan of either tbh. Wdyt?

Reply:
lldb really only should have an opinion about the selected frame index when a process stops, before control gets handed back to the user.  After that, it's up to them to change it whenever they feel like it.  So maybe we don't need to be as protective of this ivar?